### PR TITLE
Final PR Packages Upgrade

### PR DESCRIPTION
**Pull request by DeeptanshuPrasad**
> Final PR For Packages Upgrade

### CODE REVIEW

File: pages/blog/[...slug].vue [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+const fetchRelatedArticles = async () => {
+  const result = await queryContent('blog').find()
+  relatedArticles.value = result.filter(
+    (relatedArticle) => article.value._path !== relatedArticle._path
+  )
+  const randIndex = Math.floor(
+    Math.random() * (relatedArticles.value.length - 2)
+  )
+  relatedArticles.value = relatedArticles.value.slice(randIndex, randIndex + 2)
+}
```
Comment (piyushsinghania) [inline pages/blog/[...slug].vue:326]: USe it  through a utility file

File: pages/features/collaboration.vue [hunk 2]
```diff
@@ -109,6 +109,218 @@ export default {
...
+    const fetchRandomTestimonials = async () => {
+      try {
+        const testimonials = await allTestimonials
+        const randIndex = Math.floor(Math.random() * (testimonials.length - 2))
+        randomTestimonials.value = testimonials.slice(randIndex, randIndex + 2)
+      } catch (error) {
+        console.error('Error fetching random testimonials:', error)
+      }
+    }
+
```
Comment (piyushsinghania) [inline pages/features/collaboration.vue:312]: use it through the utility method.

File: pages/templates/categories/[...slug].vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    :templates="templates"
+    :templateCategories="categories"
+  />
+</template>
+
+<script setup>
+import { ref, onMounted, computed } from 'vue'
+import getSiteMeta from '@/utils/getSiteMeta'
+import Templates from '@/components/template/Templates.vue'
+// import { useRoute } from 'vue-router';
```
Comment (piyushsinghania) [inline pages/templates/categories/[...slug].vue:13]: Remove this line if not needed

File: pages/use-case/software-and-technology.vue [hunk 4]
```diff
@@ -1,354 +1,360 @@
...
+import AdvanceFeatures from '../../components/use-case/advance-features.vue'
+import IndustrySpecificUseCase from '../../components/use-case/industry-specific-use-case.vue'
+import FormBuildersComparisionTable from '../../components/form-builders-comparision-table.vue'
+
+import ThreeStepsCreateForm from '../../components/ThreeStepsCreateForm.vue'
+import Testimonial from '../../components/Testimonial.vue'
+import { allTestimonials } from '@/constants/testimonials'
+import TemplateSection from '../../components/TemplateSection.vue'
+import CallToActionSection from '@/components/CallToActionSection.vue'
+import FAQwithCategories from '../../components/FAQwithCategories.vue'
```
Comment (piyushsinghania) [inline pages/use-case/software-and-technology.vue:65]: Can we update the paths of the components to be absolute using `@`

**Review by piyushsinghania — COMMENTED** [review]
Reviewed files other than media files (.png, .svg, .jpg, .mp4, fonts)

File: nuxt.config.js [hunk 5]
```diff
@@ -91,31 +97,20 @@ export default {
   ],
 
   // Modules: https://go.nuxtjs.dev/config-modules
-  modules: [
-    '@nuxtjs/robots',
-    '@nuxtjs/axios',
```
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply nuxt.config.js:96]: Yes, we don't need axios now

File: pages/blog/index.vue [hunk 6]
```diff
@@ -29,46 +29,53 @@ export default {
...
-    const heroArticles = await $content('blog')
-      .where({
-        published: true,
-        featured: true,
-      })
-      .sortBy('createdAt', 'desc')
-      .fetch()
+    const { data: heroArticles } = useAsyncData(
+      async () => {
+        return await queryContent('/blog')
```
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply pages/blog/index.vue:44]: We need to use the useAsyncData and return for fetching the articles through `queryContent` by Nuxt 3.

File: pages/blog/[...slug].vue [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+      'Form builder showing drag and drop functionality',
+  }
+  return getSiteMeta(metaData)
+})
+
+useHead({
+  title: article?.metaTitle,
+  meta: [
+    { meta },
+    {
```
Comment (piyushsinghania) [inline pages/blog/[...slug].vue:210]: meta is an array of objects, don't we need to destructure?

**Review by piyushsinghania — APPROVED** [review]
Approved! Can we just test it once in preview mode before deploying @DeeptanshuPrasad @ankitsinghaniyaz?

File: pages/templates/categories/[...slug].vue [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+      url: `https://formester.com/templates/${template.slug}`,
+      name: template.name,
+      image: template.previewImageUrl,
+      description: template.description,
+    }
+  })
+})
+
+useHead({
+  meta: [meta.value],
```
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply pages/templates/categories/[...slug].vue:83]: @piyushsinghania yes, here in this case, meta.value is not an array. Its an object, hence we are assigning it inside an [] which will appear like this [{}].

File: plugins/notifications-client.js
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply plugins/notifications-client.js:1]: Added this plugin as well.

File: pages/blog/[...slug].vue (hunk 7 above)
Comment (DeeptanshuPrasad) [reply pages/blog/[...slug].vue]: This is resolved

File: pages/templates/categories/[...slug].vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply pages/templates/categories/[...slug].vue]: This is resolved. We need to use only `meta` which is returning the value in [] format only.

File: components/template/PreviewModal.vue [hunk 9]
```diff
@@ -10,15 +10,15 @@
 
     <div class="preview-heading-container">
       <span class="preview-heading">
-        {{ template.name }}
+        {{ template?.name }}
```
[the remark this answers is not in the document]
Comment (DeeptanshuPrasad) [reply components/template/PreviewModal.vue:13]: This `?` is now removed using `useAsyncData` composable for fetching the data.

**Review by piyushsinghania — APPROVED** [review]
Left few comments on some files and approved

**Pull request closed without merging**